components/_impl/workers/subprocess_rpc.py

```diff
@@ -162,22 +169,25 @@ def maybe_timeout_read(cls, pipe: "Pipe") -> None:
 
 
 class Pipe:
-    """Helper class to move data in a robust fashon.
+    """Helper class to move data in a robust fashion.
 
     This class handles:
-        1) File descriptor lifetimes
-        2) File descriptor inheritance
-        3) Message packing and unpacking
-        4) (Optional) timeouts for reads
+        1) Process liveness checks
+        2) File descriptor lifetimes
+        3) File descriptor inheritance
+        4) Message packing and unpacking
+        5) (Optional) timeouts for reads
     """
 
     def __init__(
         self,
+        proc_id: int = 0,
```

Comment:
nit: Should probably call this `writer_pid` since a pipe connects two processes. Also this should be an `Optional[int] = None` so `get_pid()` can assert that it was set.